build-contract

```diff
@@ -1,16 +1,66 @@
-#!/bin/bash
-set -e
-if [[ ! -z "$DEBUG" ]]; then
-  set -x
-fi
+#!/usr/bin/env bash
+[ -z "$DEBUG" ] || set -x
+set -eo pipefail
 trap "exit" INT
 
 ROOT=$PWD
 DIR=`dirname $(realpath $0)`
-if [[ "$1" == "push" ]]; then
-  DO_PUSH=true
-else
-  DO_PUSH=false
+
+# Default values
+DO_PUSH=false
+BUILD_CONTRACT_DRYRUN=false
+BUILD_CONTRACT_GREP=""
+BUILD_CONTRACT_EXTRA=""
+
+# Help function
+function show_help {
+  echo "Usage: $0 [options]"
+  echo "Options:"
+  echo "  push                   Push images after build"
+  echo "  --dry-run              Print commands instead of executing them"
```

Comment:
Nice idea. It's a bit hard to see from the diff which docker-compose commands will be run and not while in this mode, but I guess you were happy with the behavior when testing 😄